web/stripe-template/api/routes/webhook/POST-subscription.js

```diff
@@ -0,0 +1,144 @@
+import { RouteContext } from "gadget-server";
+import { getStripeWebhookEvent } from "../../stripe";
+import { objKeyConvert, destructure } from "../../utils";
+
+/**
+ * Route handler for POST webhook/subscription
+ *
+ * @param { RouteContext } route context - see: https://docs.gadget.dev/guides/http-routes/route-configuration#route-context
+ *
+ */
+export default async function route({
+  request,
+  reply,
+  api,
+  logger,
+  connections,
+}) {
+  let event = request.body;
+
+  try {
+    event = getStripeWebhookEvent({
+      logger,
+      request: request,
+      endpointSecret: process.env.STRIPE_SUBSCRIPTION_WEBHOOK_SECRET,
+    });
+  } catch (err) {
+    return await reply.status(400).send();
+  }
+
+  /**
+   * Notes:
+   * objKeyConvert is used to convert snake_case data coming from Stripe to camelCase. Date/time fields are converted, and stripe "id" fields are changed to "stripeId"
+   * destructure is used to remove additional fields from events
+   */
+
+  let subscription;
+  // Handle the event
+  switch (event.type) {
+    case "customer.subscription.created":
+      subscription = destructure({
+        topic: "subscription",
+        obj: objKeyConvert(event.data.object),
+      });
+
+      logger.info({ subscription }, "Subscription created.");
+
+      // if there is an active user for this customer, then add the relationship to this subscription
+      const user = await api.user.maybeFindFirst({
+        filter: { stripeCustomerId: { equals: subscription.customer } },
+        select: { id: true },
+      });
+
+      if (user) {
+        subscription.user = {
+          _link: user.id,
+        };
+      }
+
+      // call the stripeSubscription.create action to create a new subscription record
+      try {
+        await api.stripe.subscription.create(subscription);
+
+        logger.info(
+          { subscription },
+          `Subscription status is ${subscription.status}.`
+        );
+      } catch (e) {
+        logger.warn("Subscription already exists");
+      }
+
+      break;
+
+    case "customer.subscription.updated":
+      subscription = destructure({
+        topic: "subscription",
+        obj: objKeyConvert(event.data.object),
+      });
+
+      await api.stripe.subscription.upsert({
+        on: ["stripeId"],
+        ...subscription,
+      });
+
+      break;
+    case "customer.subscription.deleted":
+      subscription = destructure({
+        topic: "subscription",
+        obj: objKeyConvert(event.data.object),
+      });
+
+      const subscriptionToDelete = await api.stripe.subscription.maybeFindFirst(
+        {
+          filter: { stripeId: { equals: subscription.stripeId } },
+          select: { id: true },
+        }
+      );
+
+      if (subscriptionToDelete) {
+        await api.internal.stripe.subscription.delete(subscriptionToDelete.id);
```

Comment:
Cancelling a subscription doesn't quite work. As a user, I still see the price I previously selected as active on the billing page. I think we need to remove the priceId on the user model on delete as well

Reply:
Fixing it

Reply:
Fixed with the removal of the field